v2/yarpcthrift/thriftrw-plugin-yarpc2/server.go

```diff
@@ -111,13 +113,18 @@ func (h handler) <.Name>(ctx <$context>.Context, body <$wire>.Value) (<$yarpcthr
 	<end>
 
 	result, err := <$prefix>Helper.WrapResponse(<if .ReturnType>success,<end> appErr)
+	if err != nil {
+		return nil, err
+	}
 
-	var response <$yarpcthrift>.Response
-	if err == nil {
-		response.Exception = appErr
-		response.Body = result
+	if appErr != nil {
+		return nil, <$yarpcerror>.New(
+			<$yarpcerror>.CodeUnknown,
+			appErr.Error(),
```

Comment:
We might still want this for logging purposes, because once we're in the obs middleware, we've already taken the error details and placed it in the response buffer.